### PR TITLE
Factored up RAML docs into common document and updated many endpoints.

### PR DESCRIPTION
**Pull request by tylercloke**
Including:
+ Complete documentation for users endpoint.
+ Complete documentation for clients endpoint.
+ Complete documentation for keys endpoint.
+ Pulled all documentation into base doc.

Ping @chef/lob

### CODE REVIEW

File: src/oc_erchef/raml-docs/README.md [hunk 1]
```diff
@@ -14,5 +14,6 @@ plans to make these docs official or comprehensive.
 
 1) swallow your pride and get node.js
```
Comment (joedevivo) [inline src/oc_erchef/raml-docs/README.md]: lol

**Comment by joedevivo** [comment]
checked it out and followed the readme. was rewarded with docs!

File: src/oc_erchef/raml-docs/README.md (hunk 1 above)
Comment (tylercloke) [inline src/oc_erchef/raml-docs/README.md]: I think that was @danielsdeleo :P

Comment (danielsdeleo) [inline src/oc_erchef/raml-docs/README.md]: indeed

**Comment by joedevivo** [comment]
+1 on raml implementation. I still need to come back and review the content.

**Comment by joedevivo** [comment]
+1 on content, short of verifying every attribute in the code.

@marcparadise should probably look at the keys api docs

**Comment by tylercloke** [comment]
I'm pretty confident in these changes (except policy, I didn't mess with that). But I loaded up chef-shell against 12.1 and literally ran all of these requests by hand to verify everything was dapper :)

**Comment by danielsdeleo** [comment]
I need to update the policies stuff. Can either collaborate on your branch or just do a PR after you merge.

**Comment by joedevivo** [comment]
+1 get it merged!

File: src/oc_erchef/raml-docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+all: index.html
+
+NPM = $(shell which npm)
+RAML2HTML = $(shell which raml2html)
+
+index.html:
+ifeq ($(NPM),)
+			@echo "You don't have node.js installed. Swallow your pride and come back later"
```
Comment (marcparadise) [inline src/oc_erchef/raml-docs/Makefile]: Recommend "node.js is required, please install it before continuing."

File: src/oc_erchef/raml-docs/README.md [hunk 3]
```diff
@@ -12,7 +12,20 @@ plans to make these docs official or comprehensive.
 
 ## Building HTML Docs From RAML Source
 
-1) swallow your pride and get node.js
-2) `npm install -g raml2html`
-3) `raml2html SOURCE_FILE.yml > OUTFILE.html`
+### Prerequisites
 
+`raml2html` is a node.js joint, so swallow your pride and get node.js
```
Comment (marcparadise) [inline src/oc_erchef/raml-docs/README.md]: Again recommend removing the 'swallow your pride' bit :)

File: src/oc_erchef/raml-docs/client-keys.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    View and add keys for the named client in an organization.
+  uriParameters:
+    key:
+      displayName: key
+      description: Name of the key owned by the client.
+      type: string
+  get:
+    is: [ versioned ]
+    description: |
+      Retrieves the specified public key and it's associated data.
```
Comment (marcparadise) [inline src/oc_erchef/raml-docs/client-keys.yml]: its associated

File: src/oc_erchef/raml-docs/policies.yml [hunk 5]
```diff
@@ -34,8 +30,8 @@ documentation:
       displayName: group - Policy Group Name
       type: string
   get:
-    description: |
-      Retrieve the Policy document
+    is: [ versioned ]
```
Comment (marcparadise) [inline src/oc_erchef/raml-docs/policies.yml]: There are no versioned behaviors for policy?

File: src/oc_erchef/raml-docs/README.md (hunk 3 above)
Comment (joedevivo) [inline src/oc_erchef/raml-docs/README.md]: It was there before, and not mine to remove :P, you'll have to take it up with @danielsdeleo

**Comment by marcparadise** [comment]
:+1: outside of the minor comments

File: src/oc_erchef/raml-docs/README.md (hunk 3 above)
Comment (danielsdeleo) [inline src/oc_erchef/raml-docs/README.md]: It's fine to nuke it. This probably won't go over as well with node.js folks as it does with erlang folks

File: src/oc_erchef/raml-docs/policies.yml (hunk 5 above)
Comment (tylercloke) [inline src/oc_erchef/raml-docs/policies.yml]: Idk I thought literally all of our endpoints returned the version header? That's all that means. It returns the version header.

Comment (marcparadise) [inline src/oc_erchef/raml-docs/policies.yml]: Ah ok, cool

**Pull request merged**